### ISSUE TITLE
Recurrent error with RELAX

### ISSUE THREAD

**Issue by flallema** [veg/hyphy#813]
Dear Datamonkey team,

I was using RELAX one month ago without problem. Now I'm not able to run any analysis and I receive the following error message : We're sorry, there was an error processing your job. Please try again, or visit our GitHub issues and create an issue if the issue persists.

I tried different nexus and fasta data files, including the example file HIV-1 POL, and different test / reference settings with no success. Below are one data file (nexus) I use and the associated test vs reference grouping.

[noycf1.txt](https://github.com/veg/hyphy/files/2144880/noycf1.txt)

![partition_m1](https://user-images.githubusercontent.com/40661037/42029416-c86dec8a-7acf-11e8-907f-2c419e08d512.png)

I'm using Windows 7 pro and Firefox 61.0 (64 bits)

Thank you for your job and the development of such valued tools!
félix

**Comment by stevenweaver** [comment]
Dear @flallema, 

This is indeed a bug. HyPhy is automatically detecting the tree within the uploaded NEXUS file. The tree within the NEXUS file itself is never annotated. Datamonkey creates a new Newick file with the appropriate annotations. 

For FASTA files, we strip the Newick tree from the FASTA file for HyPhy processing. However, it appears that we are not doing this for NEXUS files. 

Best,
Steven

**Comment by stevenweaver** [comment]
Dear @flallema, 

I was a bit quick with my assumption. It appears that you have a malformed NEXUS file. `ENDS;` should be `END;` after the `TREE` definition. 

Best,
Steven

**Comment by flallema** [comment]
Dear @stevenweaver ,

Thank you very much for your quick answer. I removed the "S" at the "END" for the NEXUS files and unfortunately, the problem persists. But now my FASTA files are working, I'm just not able to run the program with my own tree. 

More precisely, when I add a Newick tree line at the end of the FASTA file, RELAX recognizes it, the tree's topology is good, but branch lengths are not computed.

![relax_tree](https://user-images.githubusercontent.com/40661037/42079562-506c6890-7b80-11e8-8cbd-7ca90c7c96fc.png)

Maybe there is a problem in the way I write the tree in the FASTA file? (attached below) Or maybe I missed something in your answer.

[noycf_results_tree.txt](https://github.com/veg/hyphy/files/2148251/noycf_results_tree.txt)

Best
Félix

**Comment by stevenweaver** [comment]
Dear @flallema, 

Our FASTA and NEXUS parser strips branch lengths from the supplied Newick Tree. 

Please note @sjspielman's comments at [hyphy.org](http://hyphy.org/methods/general/#mg94xrev-framework)
> Most methods will perform a global MG94xREV fit to optimize branch length and nucleotide substitution parameters before proceeding to hypothesis testing. Several methods (FEL, FUBAR, and MEME) additionally pre-fit a GTR nucleotide model to the data, using the estimated parameters as starting values for the global MG94xREV fit, as a computational speed-up. Resulting branch length and nucleotide substitution parameters are subsequently used as initial parameter values during model fitting for hypothesis testing.

RELAX is not an exception. It performs both preliminary GTR nucleotide model fits and a global MG94xREV fit for branch length optimizations.

With that said, the Neighbor Joining tree and the User supplied tree should exhibit the same behavior. I will leave the issue open and discuss with the lab what the right approach should be.

Best,
Steven

**Comment by flallema** [comment]
Dear Steven,

Thank you for the explanations. I carried out some analyses in order to check whether choosing user or NJ tree gives different results. The answer seems to be yes in some cases.

I performed three tests, trying to follow a "nesting" approach. The test / reference design I used for these tests are in the picture below. It shows how I dealt with the NJ tree computed by the program. I'm however not sure that my choices regarding which branches should be considered as test or reference in the NJ tree are very legal (ie, do they really correspond to what I have done for the user tree)

![design_tests](https://user-images.githubusercontent.com/40661037/42220361-ff7f1b74-7ece-11e8-8877-aa0fec20a412.png)

The results for different sets of genes are below

![relax_results](https://user-images.githubusercontent.com/40661037/42220558-a0bcbf96-7ecf-11e8-922e-93ff18617032.png)

It also shows for comparison what I obtained with a former PAML analysis, testing different w for the same branches as labelled above. A major disagreement appears for rbcL, test n°2 (M2): the user tree yielded strong evidence for relaxation, whereas it was an increase in selection strength for the NJ tree. Could it be due to differences in what is considered as plesiomorphic states? Another disagreement for housekeeping genes, test n°2.

I hope this may help you in your discussion. I would be pleased to have your feedbacks!

Have a nice day
Félix
